core/internal/evaluator/caching.go

```diff
@@ -281,7 +281,7 @@ func evaluatePartitionStatus(partition *protocol.ConsumerPartition, minimumCompl
 	}
 
 	// Slice the offsets to remove all nil entries (they'll be at the start)
-	firstOffset := len(partition.Offsets) - 1
+	firstOffset := len(partition.Offsets) // defaults to the length, so if all offsets are nil we make an empty slice
```

Comment:
Without this change, a runtime error: invalid memory address or nil pointer dereference was happening.